gunrock/oprtr/1D_oprtr/for_all.cuh

```diff
@@ -50,6 +50,36 @@ __global__ void ForAll_Kernel(ArrayT array, ApplyLambda apply, SizeT length) {
   }
 }
 
+template <typename ArrayT, typename SizeT, typename ApplyLambda>
+__global__ void ForAllDebug_Kernel(ArrayT array, ApplyLambda apply, SizeT length) {
+  // typedef typename ArrayT::SizeT SizeT;
+  const SizeT STRIDE = (SizeT)blockDim.x * gridDim.x;
+  SizeT i = (SizeT)blockDim.x * blockIdx.x + threadIdx.x;
+  while (i < length) {
+/*    if (blockDim.x * blockIdx.x + threadIdx.x == 0){
+        printf("%d points done\n", i);
+    }*/
+    apply(array + 0, i);
+    i += STRIDE;
+  }
+}
```

Comment:
Why does debug kernel exist here? If this is solely used for debugging, it should be a test. I don't think it should be added to the main API.